src/mono/mono/metadata/threads-types.h

```diff
@@ -580,4 +580,10 @@ mono_threads_summarize_execute (MonoContext *ctx, gchar **out, MonoStackHash *ha
 gboolean
 mono_threads_summarize_one (MonoThreadSummary *out, MonoContext *ctx);
 
+#if SIZEOF_VOID_P == 4
+/* Spin lock for unaligned InterlockedXXX 64 bit functions on 32bit platforms. */
+#define mono_interlocked_lock() mono_os_mutex_lock (&interlocked_mutex)
+#define mono_interlocked_unlock() mono_os_mutex_unlock (&interlocked_mutex)
```

Comment:
```suggestion
extern mono_mutex_t mono_interlocked_mutex;
static inline void
mono_interlocked_lock(void) {
	mono_os_mutex_lock (&mono_interlocked_mutex);
}
static inline void mono_interlocked_unlock() {
	mono_os_mutex_unlock (&mono_interlocked_mutex);
}
```

1. You need a declaration for the global variable.  (just having a definition in threads.c is not enough).
2. Global variables should have a `mono_` prefix
3. No need to use `#define` when a `static inline` function will work since we're not doing any preprocessor stuff in the implementation.
4. You will need to go and change the places in `threads.c` that `init and `destory` the mutex to use the new name

Reply:
I declared the global variable with extern at the location where I used it again. And it did worked. I didn't add the declaration in the header file, because I wanted to limit its usage, since it was a static var before.